File: app/src/main/java/com/example/keinomoto/automatchingsearchapp/SearchResultAdapter.java
package com.example.keinomoto.automatchingsearchapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class SearchResultAdapter extends BaseAdapter {

    Context context;
    LayoutInflater layoutInflater = null;
    List<SearchInfo> searchInfoList;

    public SearchResultAdapter(Context context) {
        this.context = context;
        this.layoutInflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    public void setSearchInfoList(List<SearchInfo> searchInfoList) {
        this.searchInfoList = searchInfoList;
    }

    @Override
    public int getCount() {
        return searchInfoList.size();
    }

    @Override
    public Object getItem(int position) {
        return searchInfoList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return searchInfoList.get(position).getSearchId();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        convertView = layoutInflater.inflate(R.layout.scrapingrow,parent,false);

        ((TextView)convertView.findViewById(R.id.title)).setText(searchInfoList.get(position).getTitle());

        return convertView;
    }
}

File: app/src/main/java/com/example/keinomoto/automatchingsearchapp/SearchResultAdapter.kt
package com.example.keinomoto.automatchingsearchapp

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import java.util.ArrayList

class SearchResultAdapter(internal var context: Context) : BaseAdapter() {
    internal var layoutInflater: LayoutInflater? = null
    internal var searchInfoList = mutableListOf<SearchInfo>()

    init {
        this.layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    fun setSearchInfoList(searchInfoList: MutableList<SearchInfo>) {
        this.searchInfoList = searchInfoList
    }

    override fun getCount(): Int {
        return searchInfoList.size
    }

    override fun getItem(position: Int): Any {
        return searchInfoList[position]
    }

    override fun getItemId(position: Int): Long {
        return searchInfoList[position].searchId!!.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        convertView = layoutInflater!!.inflate(R.layout.scrapingrow, parent, false)

        (convertView.findViewById<View>(R.id.title) as TextView).text = searchInfoList[position].title

        return convertView
    }
}
